### PR TITLE
Documentation/attachments

### PR DESCRIPTION
**Pull request by wow-such-code**
- adapt script to the version currently used on production (different datastore server)
- add readme file with expected data structure explanation

### CODE REVIEW

**Comment by wow-such-code** [comment]
[register-attachments-dropbox ETL readme](https://trello.com/c/tETzIfWm/538-register-attachments-dropbox-etl-readme)

File: drop-boxes/register-attachments-dropbox/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+|-<anything> (top level folder name, normally a time stamp of upload time)
+    |
+    |- metadata.txt
+```
+
+openBIS structure overview:
+
+Attachments are attached to the Q_PROJECT_DETAILS experiment type and its sample type Q_ATTACHMENT_SAMPLE.
+
+TODO: ER model.
```
Comment (jenniferboedker) [inline drop-boxes/register-attachments-dropbox/README.md]: Is it planned to add that here in this PR?

File: drop-boxes/register-attachments-dropbox/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+```
+|-<anything> (top level folder name, normally a time stamp of upload time)
+    |
+    |- metadata.txt
+```
+
+openBIS structure overview:
+
+Attachments are attached to the Q_PROJECT_DETAILS experiment type and its sample type Q_ATTACHMENT_SAMPLE.
```
Comment (jenniferboedker) [inline drop-boxes/register-attachments-dropbox/README.md]: maybe add the sample code schema here in regex? (in the code its Q\w{4}000, this is python specific I guess?)

File: drop-boxes/register-attachments-dropbox/register-attachment-dropbox.py [hunk 3]
```diff
@@ -84,26 +84,14 @@ def process(transaction):
...
 	space = sa.getSpace()
 	if not attachmentReady:
-		expID = '/' + space + '/' + project + '/'+ project+'_INFO'
-		exp = transaction.getExperimentForUpdate(expID)
-		if not exp:
-			exp = transaction.createNewExperiment(expID, "Q_PROJECT_DETAILS")
+		infoSampleID = "/"+space+"/"+code
+		sa = transaction.getSampleForUpdate(infoSampleID)
+	if not sa:
+		exp = transaction.createNewExperiment('/' + space + '/' + project + '/'+ project+'_INFO', "Q_PROJECT_DETAILS")
```
Comment (jenniferboedker) [inline drop-boxes/register-attachments-dropbox/register-attachment-dropbox.py]: why did you change this? Is this related to the documentation?

**Review by jenniferboedker — CHANGES_REQUESTED** [review]
Looks good, I added a few questions where it was not clear to me what happens here/if there is some further work required.

File: drop-boxes/register-attachments-dropbox/README.md (hunk 1 above)
Comment (wow-such-code) [reply drop-boxes/register-attachments-dropbox/README.md]: good question, let's ask the team

File: drop-boxes/register-attachments-dropbox/register-attachment-dropbox.py (hunk 3 above)
Comment (wow-such-code) [reply drop-boxes/register-attachments-dropbox/register-attachment-dropbox.py]: Unfortunately not. The script is mainly used on another datastore server (that is not kept up to date with this repository, since it only contains this script...long story), so I wanted to make sure that the Github documentation is referencing the version currently used in production.

Comment (jenniferboedker) [reply drop-boxes/register-attachments-dropbox/register-attachment-dropbox.py]: I think it would be better to make this change in another PR to make changes more explicitly visible in the commit history. Can you imagine to create a small PR for that? I volunteer to review it fast ;)

Comment (wow-such-code) [reply drop-boxes/register-attachments-dropbox/register-attachment-dropbox.py]: Good idea, I will do that!

File: drop-boxes/register-attachments-dropbox/README.md (hunk 2 above)
Comment (wow-such-code) [reply drop-boxes/register-attachments-dropbox/README.md]: regex added

File: drop-boxes/register-attachments-dropbox/README.md (hunk 1 above)
Comment (wow-such-code) [reply drop-boxes/register-attachments-dropbox/README.md]: issue opened, todo removed

**Review by jenniferboedker — APPROVED** [review]
